src/openvpn/options.c

```diff
@@ -6556,7 +6556,7 @@ add_option(struct options *options,
                 msg(msglevel, "route-ipv6 parameter network/IP '%s' must be a valid address", p[1]);
                 goto err;
             }
-            if (p[2] && !ipv6_addr_safe(p[2]))
+            if (p[2] && !ipv6_addr_safe(p[2]) && !ipv6_is_special_addr(p[2]))
```

Comment:
```suggestion
            if (p[2] && (!ipv6_addr_safe(p[2]) || !ipv6_is_special_addr(p[2])))
```
The logic looks wrong to me. The error should be trigger if either check is failing.

Reply:
Why? When p[2] is "net_gateway_ipv6" ipv6_addr_safe() will return false, but ipv6_is_special_addr() will return true. Which is exactly what we want to check. The error is triggered only if p[2] is garbage.